src/stateless_validation.cpp

```diff
@@ -414,4 +414,42 @@ bool checkPopData(const PopData& popData, ValidationState& state) {
   return true;
 }
 
+bool checkPopData(PopValidator& validator,
+  const PopData& popData,
+  ValidationState& state) {
+
+  std::vector<std::future<ValidationState>> results;
+  results.reserve(popData.context.size() + popData.vtbs.size() +
+                    popData.atvs.size());
+
+  for (const auto& b : popData.context) {
+    results.push_back(validator.addCheck(b));
+  }
+  for (const auto& vtb : popData.vtbs) {
+    results.push_back(validator.addCheck(vtb));
+  }
+  for (const auto& atv : popData.atvs) {
+    results.push_back(validator.addCheck(atv));
+  }
+
+  size_t index = 0;
+  for (auto& r : results) {
+    auto result = r.get();
+    if (result.IsInvalid()) {
```

Comment:
```suggestion
    if (!result.IsValid()) {
```
because there's also `IsError()`